completion/api/v1/views.py

```diff
@@ -148,3 +148,37 @@ def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
             }, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
         return Response({"detail": _("ok")}, status=status.HTTP_200_OK)
+
+
+class MilestonesExperimentView(APIView):
+    """
+    Handles API endpoints for the milestones experiments.
+    """
+
+    permission_classes = (permissions.IsAuthenticated, IsUserInUrl)
+    REQUIRED_KEYS = ['course_key', 'subsection_id']
+
+    def get(self, request, username):
+        """
+        Returns completion for a (user, subsection, course).
+
+        REST Endpoint format:
+        {
+            "course_key": course_key,
+            "subsection_id": subsection_id
+        }
+        The username should be passed in through the URL.
+        """
+        for key in self.REQUIRED_KEYS:
+            if key not in request.data:
+                raise ValidationError(_("Key '{key}' not found.".format(key=key)))
+
+        user = username
+        course_key = request.data['course_key']
+        subsection_id = request.data['subsection_id']
+
+        try:
+            completion = BlockCompletion.objects.get(user=user, course_key=course_key, block_key=subsection_id)
```

Comment:
There aren't `BlockCompletion` records for subsections - we only store completion data on leaf-level blocks (components).  So, if you want to know whether a subsection is complete, you're going to have to recursively determine if each of it's children (which are presumably verticals) is complete.  You could also just change this to do what the `vertical_is_complete()` function in services.py does, and have the caller hit the endpoint for each vertical in the subsection.